data/kernels/blendop.cl

```diff
@@ -146,100 +159,115 @@ typedef enum dt_dev_pixelpipe_display_mask_t
   DT_DEV_PIXELPIPE_DISPLAY_HSL_H = 10 << 3,
   DT_DEV_PIXELPIPE_DISPLAY_HSL_S = 11 << 3,
   DT_DEV_PIXELPIPE_DISPLAY_HSL_l = 12 << 3,
+  DT_DEV_PIXELPIPE_DISPLAY_JzCzhz_Jz = 13 << 3,
+  DT_DEV_PIXELPIPE_DISPLAY_JzCzhz_Cz = 14 << 3,
+  DT_DEV_PIXELPIPE_DISPLAY_JzCzhz_hz = 15 << 3,
   DT_DEV_PIXELPIPE_DISPLAY_ANY = 0xff << 2
 } dt_dev_pixelpipe_display_mask_t;
 
 
 float
-blendif_factor_Lab(const float4 input, const float4 output, const unsigned int blendif, global const float *parameters, const unsigned int mask_mode, const unsigned int mask_combine)
+blendif_factor_Lab(const float4 input, const float4 output,
+                   const unsigned int blendif, global const float *parameters,
+                   const unsigned int mask_mode, const unsigned int mask_combine)
 {
-  float result = 1.0f;
   float scaled[DEVELOP_BLENDIF_SIZE];
 
   if(!(mask_mode & DEVELOP_MASK_CONDITIONAL)) return (mask_combine & DEVELOP_COMBINE_INCL) ? 0.0f : 1.0f;
 
-  scaled[DEVELOP_BLENDIF_L_in] = clamp(input.x / 100.0f, 0.0f, 1.0f);			// L scaled to 0..1
-  scaled[DEVELOP_BLENDIF_A_in] = clamp((input.y + 128.0f)/256.0f, 0.0f, 1.0f);		// a scaled to 0..1
-  scaled[DEVELOP_BLENDIF_B_in] = clamp((input.z + 128.0f)/256.0f, 0.0f, 1.0f);		// b scaled to 0..1
+  scaled[DEVELOP_BLENDIF_L_in] = input.x / 100.0f,			// L scaled to 0..1
+  scaled[DEVELOP_BLENDIF_A_in] = input.y / 128.0f;		// a scaled to -0.5..0.5
```

Comment:
doesn't that break compatibility with older edits ?

Reply:
I hope not...

This change has to be reviewed together with the code in function `dt_develop_blend_process_cl` in blend.c.

1. Removed clamping to allow setting mask limits on values > 1.0f (although this is probably not valid from the color space point of view). The (open) limits (0.0f and 1.0f) are converted to -INFINITY and INFINITY. A value clamped to one of the limits before is now still in one of the ranges when the parametric mask parameters are open-ended.

2. Removed the offset, because it is now also handled by the pre-processing in blend.c (instead of changing the pixel values, the limits are changed).